backend/src/modules/urls/url.controller.ts

```diff
@@ -1,31 +1,58 @@
-import {Body, Controller, Delete, Get, Param, Post} from '@nestjs/common';
+import {Body, Controller, Delete, Get, Param, Post, Res} from '@nestjs/common';
+import {Response} from 'express';
 import {UrlService} from './url.service';
 import {UrlSimpleDto} from './dto/url-simple.dto';
 import {UrlRequestDto} from './dto/url-request.dto';
+import {RequireLoggedIn} from '../../guards/role-container';
+import {AuthUser} from "../../decorator/auth-user.decorator";
+import {UserEntity} from '../user/entities/user.entity';
 
 @Controller('urls')
 export class UrlController {
     constructor(private readonly urlService: UrlService) {}
 
+    @RequireLoggedIn()
     @Post()
-    async create(@Body() body: UrlRequestDto): Promise<UrlSimpleDto> {
-        const toRequest = UrlRequestDto.toUrlRequest(body);
-        return UrlSimpleDto.fromDomain(await this.urlService.createUrl(toRequest));
+    async create(
+        @Body() body: UrlRequestDto,
+        @AuthUser() user: UserEntity
+    ): Promise<UrlSimpleDto> {
+        const url = await this.urlService.createUrl(
+            UrlRequestDto.toUrlRequest(body),
+            user
+        );
+        return UrlSimpleDto.fromDomain(url);
     }
 
-    @Get(':shortCode')
-    async getUrl(@Param('shortCode') shortCode: string): Promise<UrlSimpleDto> {
-        return UrlSimpleDto.fromDomain(await this.urlService.getUrl(shortCode));
+    @RequireLoggedIn()
+    @Get()
+    async getAllUrls(@AuthUser() user: UserEntity): Promise<UrlSimpleDto[]> {
+        const urls = await this.urlService.getAllUrls(user);
+        return urls.map(UrlSimpleDto.fromDomain);
```

Comment:
Thêm hàm fromDomains bên dto class nha

```suggestion
        return UrlSimpleDto.fromDomains(await this.urlService.getAllUrls(user));
```